lib/storage/clusterconfig/clusterconfig.go

```diff
@@ -227,6 +227,8 @@ type Spec struct {
 type ComponentConfigs struct {
 	// Kubelet defines kubelet configuration
 	Kubelet *Kubelet `json:"kubelet,omitempty"`
+	// GravityControllerService defines gravity-site service configuration
+	GravityControllerService `json:"gravityControllerService,omitempty"`
```

Comment:
You will not be able to take advantage of `omitempty` unless this is a primitive with a known zero-value (incl. a pointer).